### PR TITLE
feat(sbb-checkbox): add size xs

### LINKED ISSUE #2767: Checkbox: Size XS
https://www.figma.com/file/9r6xSfNmEfCFxl1yFYedrj/Lyne-Components?type=design&node-id=29393%3A203337&mode=design&t=y66AzDZgLd66BKFE-1

### PR DESCRIPTION
**Pull request by DavideMininni-Fincons**
## Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/sbb-design-systems/lyne-components/blob/main/docs/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/sbb-design-systems/lyne-components/blob/main/docs/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [pull request tracker](https://github.com/sbb-design-systems/lyne-components/pulls) for a Pull Request (PR) that matches the one I want to submit, without success.

## Issue

This PR Closes #2767 

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

See [Review Guidelines](../docs/REVIEW.md) for more information what is checked during review process.

## Changes

Changes in this pull request:

- the `size` property in the `sbb-chechbox` has now a new value named `xs`
- the `sbb-visual-checkbox` has now a `size` property with the same values of the sbb-checkbox's size

## Browsers

I tested the build on the following browsers:

- [ ] Firefox Desktop
- [x] Chrome Desktop
- [ ] Edge Desktop
- [ ] Safari Desktop
- [ ] Chrome Mobile
- [ ] Safari Mobile

## Screen readers

I tested the build on the following browsers:

- [ ] JAWS Firefox Desktop
- [ ] JAWS Chrome Desktop
- [ ] NVDA Firefox Desktop
- [ ] NVDA Chrome Desktop
- [ ] VoiceOver Safari Desktop
- [ ] VoiceOver Chrome Desktop
- [ ] VoiceOver Safari Mobile
- [ ] Android Accessibility Suite Chrome Mobile

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/elements/checkbox/checkbox-panel/checkbox-panel.ts [hunk 1]
```diff
@@ -51,6 +56,16 @@ export class SbbCheckboxPanelElement extends SbbPanelMixin(
     panelConnected: 'panelConnected',
   } as const;
 
+  /** Label size variant, either m or s. */
+  @property({ reflect: true })
+  public set size(value: SbbCheckboxPanelSize) {
+    this._size = value;
+  }
+  public get size(): SbbCheckboxPanelSize {
+    return this._size;
```
Comment (TomMenga) [inline src/elements/checkbox/checkbox-panel/checkbox-panel.ts:65]: I think you missed the inheritance from the group
If the group has size 'xs' I would fallback to 's' on the panel

File: src/elements/checkbox/checkbox-panel/checkbox-panel.ts [hunk 2]
```diff
@@ -16,11 +16,16 @@ import type {
...
+import {
+  checkboxCommonStyle,
+  SbbCheckboxCommonElementMixin,
+  type SbbCheckboxSize,
+} from '../common.js';
 
 import '../../screen-reader-only.js';
 import '../../visual-checkbox.js';
 
+export type SbbCheckboxPanelSize = Omit<SbbCheckboxSize, 'xs'>;
```
Comment (TomMenga) [inline src/elements/checkbox/checkbox-panel/checkbox-panel.ts:28]:
```suggestion
export type SbbCheckboxPanelSize = 's' | 'm';
```
Nitpick but I believe that extension by exclusion is generally a bad pattern

File: src/elements/checkbox/checkbox/checkbox.ts [hunk 3]
```diff
@@ -30,6 +34,16 @@ export class SbbCheckboxElement extends SbbCheckboxCommonElementMixin(
     didChange: 'didChange',
   } as const;
 
+  /** Label size variant, either m or s. */
```
Comment (TomMenga) [inline src/elements/checkbox/checkbox/checkbox.ts:37]:
```suggestion
  /** Size variant. */
```

File: src/elements/checkbox/checkbox/readme.md [hunk 4]
```diff
@@ -43,10 +43,12 @@ The component can be displayed in `disabled` or `required` state by using the se
...
 ## Style
 
-The component has two `size`, named `s` and `m` (default).
+The component has three `size`, named `xs`, `s` and `m` (default).
 
 ```html
-<sbb-checkbox value="size" size="m">Size</sbb-checkbox>
+<sbb-checkbox value="size" size="xs">Size</sbb-checkbox>
+
+<sbb-checkbox value="size" size="a">Size</sbb-checkbox>
```
Comment (TomMenga) [inline src/elements/checkbox/checkbox/readme.md:51]:
```suggestion
<sbb-checkbox value="size" size="s">Size</sbb-checkbox>
```

File: src/elements/visual-checkbox/readme.md [hunk 5]
```diff
@@ -12,13 +12,24 @@ It has four different states related to the properties `checked`, `indeterminate
...
 - a disabled state (`disabled` set to `true`): a grayed out empty square is shown.
 
+## Style
+
+The component has three `size`, named `xs`, `s` and `m` (default).
+
+```html
+<sbb-visual-checkbox size="xs"></sbb-visual-checkbox>
+
+<sbb-visual-checkbox size="a"></sbb-visual-checkbox>
```
Comment (TomMenga) [inline src/elements/visual-checkbox/readme.md:22]:
```suggestion
<sbb-visual-checkbox size="s"></sbb-visual-checkbox>
```

File: src/elements/visual-checkbox/visual-checkbox.stories.ts [hunk 6]
```diff
@@ -121,6 +130,30 @@ export const DisabledIndeterminateNegative: StoryObj = {
...
+  argTypes: defaultArgTypes,
+  args: { ...defaultArgs, size: size.options![2], indeterminate: true },
+};
+
+export const SizeXSNegative: StoryObj = {
+  render: Template,
+  argTypes: defaultArgTypes,
+  args: { ...defaultArgs, size: size.options![2], negative: true },
+};
+
```
Comment (TomMenga) [inline src/elements/visual-checkbox/visual-checkbox.stories.ts:156]: I'm conflicted here :|
We don't need combinations of params in stories anymore but maybe we should remove them once we definitively abandon chromatic.
What do you think?

**Review by TomMenga — COMMENTED** [review]
Good job 😄 
Some minor fixes

**Review by TomMenga — CHANGES_REQUESTED** [review]
I miss-clicked the previous review and published it as "comments" :|

File: src/elements/visual-checkbox/visual-checkbox.stories.ts (hunk 6 above)
Comment (DavideMininni-Fincons) [reply src/elements/visual-checkbox/visual-checkbox.stories.ts:156]: We'll remove chromatic in 6 months if I remember correctly; 
in the meantime i think we should continue to provide examples to users

**Review by TomMenga — APPROVED** [review]
LGTM 👍

**Pull request merged**